<?xml version="1.0"?>
<launch>
    <!-- ============================ SETTINGS ============================= -->

    <arg 
    name="robot_name" 
    default="my_gen3"/>

    <arg 
    name="anchor_id" 
    default="687d0663-d93e-423b-a2a5-16c9ff9b08f7"/>

    <!-- default="374594eb-6114-4816-82e4-ab69f9e2961b"/> -->

    <!-- If any of dependency or optional subfiles should not be launched (or
    are launched separately) set corresponding argument to "false". -->
    <arg 
    name="launch_teleoperation" 
    default="true"/>
    <arg 
    name="launch_controller_feedback" 
    default="true"/>

    <!-- ======================== END OF SETTINGS ========================== -->

    <!-- Teleoperation module (dependency): -->
    <group if="$(arg launch_teleoperation)">
        <include 
        file="$(find kinova_positional_control)/launch/teleoperation.launch">
            <arg 
            name="robot_name" 
            value="$(arg robot_name)"/>
        </include>
    </group>

    <!-- TF updates: -->
    <node 
    pkg="kinova_positional_control" 
    type="update_transforms.py" 
    name="tf_updates" 
    output="screen">
        <param 
        name="anchor_id" 
        value="$(arg anchor_id)"/>
        <param 
        name="robot_name" 
        value="$(arg robot_name)"/>
    </node>

    <!-- HoloLens mapping: -->
    <group ns="/$(arg robot_name)">
        <node 
        pkg="kinova_positional_control" 
        type="hololens_mapping.py" 
        name="hololens_mapping" 
        output="screen">
            <param 
            name="robot_name" 
            value="$(arg robot_name)"/>
            <param 
            name="anchor_id" 
            value="$(arg anchor_id)"/>
        </node>
    </group> 

    <!-- Robot hologram: -->
    <!-- <include 
    file="$(find kinova_positional_control)/launch/holorobot.launch">
    </include> -->
    
</launch>
